src/transport/SecureSession.cpp

```diff
@@ -104,33 +107,46 @@ void SecureSession::Reset(void)
     memset(mKey, 0, sizeof(mKey));
 }
 
-uint64_t SecureSession::GetIV(const MessageHeader & header)
+CHIP_ERROR SecureSession::GetIV(const MessageHeader & header, uint8_t * iv, size_t len)
 {
-    // The message ID is a 4 byte value. It's assumed that the security
-    // session will be rekeyed before (or on) message ID rollover.
-    uint64_t IV = header.GetMessageId() & 0xffffffff;
+    CHIP_ERROR err     = CHIP_NO_ERROR;
+    uint64_t nodeID    = 0;
+    uint32_t messageID = 0;
+
+    BufBound bbuf(iv, len);
+
+    VerifyOrExit(len >= kAESCCMIVLen, err = CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
Why is IV that is 'larger' ok? 

Thinking that if len is larger than the messag + nodeid size, the put's here will only write the rist 12 bytes anyway and leave the rest uninitialized (non deterministic?).

How about asserting the length must be exactly kAESCCMIVLen? Also a static assert of kAESCCMIVLen == sixeof(messageID) + sixeof(nodeID) may be useful.

Reply:
Static assert of ` kAESCCMIVLen == sixeof(messageID) + sixeof(nodeID)` is sort of covered by `BufBound` checks here: https://github.com/project-chip/connectedhomeip/pull/2174/files#diff-e9ea9d45465be2060c6ea896b29e252eR128